### DISCUSSION TITLE
how can i get the target node response body from upstream?

### DISCUSSION THREAD

**Discussion by Git4Future** [apache/apisix#5730] [Q&A]
i want to replace the response body from target server in upstream . 
just like string.gsub in lua. 
but i can't get the response body, ngx.arg[1] seems retun nil value. 
however, if using redirect plugin,i can only get 301 response html,not the upstream server's response

**Comment by juzhiyuan** [comment]
I only find the `response-rewrite` plugin[1], and it only supports returning a fixed value. If you need more powerful feature to replace by target text or regex, you may need to submit a PR to implement it 🤔 @tzssangglass right?

[1] https://apisix.apache.org/docs/apisix/next/plugins/response-rewrite/

**Reply by tzssangglass** [reply]
yes, releated to: https://github.com/apache/apisix/issues/5451

**Reply by Git4Future** [reply]
thks a lot
